### PR TITLE
Fix multiples call to PUT duplicates TTL entries

### LINKED ISSUE #48: duplicate "!ttl!x!timestamp|foo" entries causing unwanted deletes
My application does something like:

```javascript
db.put("foo", "xx", {ttl: ms('1h') });
```

After sometime running (12hs), I eventually end up with duplicated entries with the key format `!ttl!x!timestamp|foo` which causes an unwanted delete before the time that is expected to be deleted

I took an snapshot of my db and I ran the following command:

```
 ./leveldb_keys ~/db-snapshot/ | grep '!ttl!x!' | cut -c 23- | sort | uniq -c | grep -v '^ *1 '
   2 foo
   2 bar
   2 baz
```

I was having this issue with v2 yesterday (warning it has a different format for the key of these entries). I upgraded to v3 and I continue to have the same problem. If I leave the application running more time I eventually end up having more than 2 of these things. The application didn't crash at any time, etc.

Can you think on any kind of race condition between ttlon/off? Why ttloff might not be working on some circustance?

If someone is using this module in production with traffic, can ran the above command on test if it has duped entries will be awesome.

Thanks

### PR DESCRIPTION
**Pull request by jfromaniello**
In order to fix this problem I am using an in-memory Lock from [dominic/lock](https://github.com/dominictarr/lock)

Closes #48

### CODE REVIEW

File: level-ttl.js [hunk 1]
```diff
@@ -111,24 +112,28 @@ function ttlon (db, keys, ttl, callback) {
...
-    batchFn(
-        batch
-      , { keyEncoding: 'binary', valueEncoding: 'binary' }
-      , function (err) {
-          if (err)
-            db.emit('error', err)
-          callback && callback()
-        }
-    )
+      if (!batch.length){
```
Comment (rvagg) [inline level-ttl.js]: remove braces here

**Comment by rvagg** [comment]
@ralphtheninja this lgtm, do you want to merge and release?

**Comment by ralphtheninja** [comment]
> @ralphtheninja this lgtm, do you want to merge and release?

Will do!

**Comment by ralphtheninja** [comment]
Tests still seem a bit shaky :/

3.1.1 released

**Pull request merged**